waagent

```diff
@@ -570,6 +570,17 @@ class AbstractDistro(object):
     def mountDVD(self,dvd,location):
         return RunGetOutput(self.mount_dvd_cmd + ' ' + dvd + ' ' + location)
 
+    def fallocate(self,filename,bytes):
+        """
+        Create a non-sparse file of that size. Overwrite existing file.
+        
+        Python 3.3+: os.posix_fallocate(filename,0,bytes)
+        I am too lazy to write sys.version_info >= (3,3) here and for BSD. No, no more wrapping methods.
+        """
+        if os.path.isfile(filename)
+            os.remove(filename)
```

Comment:
If you do this, you're not *overwriting* the existing file; you're deleting it and recreating it. Those are two distinct things. (It matters if some process has an open file handle; overwrite leaves the handle pointing to a file that is still linked, but remove/create leaves that old handle pointing to a file that will be deleted as soon as the handle is closed.)